homeassistant/components/pooldose/config_flow.py

```
@@ -0,0 +1,84 @@
+            _LOGGER.debug("Connecting to PoolDose device at %s", host)
+            client = PooldoseClient(host)
+            client_status = await client.connect()
+            if client_status == RequestStatus.HOST_UNREACHABLE:
+                errors["base"] = "cannot_connect"
+            if client_status == RequestStatus.PARAMS_FETCH_FAILED:
+                errors["base"] = "params_fetch_failed"
+            elif client_status != RequestStatus.SUCCESS:
+                errors["base"] = "cannot_connect"
+            else:  # SUCCESS
+                # Successfully connected, now check API version
+                _LOGGER.debug("Connected to PoolDose device at %s", host)
+                # Check API version
+                _LOGGER.debug("Checking API version for PoolDose device at %s", host)
+                api_status, api_versions = client.check_apiversion_supported()
+                if api_status == RequestStatus.NO_DATA:
+                    errors["base"] = "api_not_set"
+                elif api_status == RequestStatus.API_VERSION_UNSUPPORTED:
+                    errors["base"] = "api_not_supported"
+                    error_placeholders = api_versions
+                else:  # SUCCESS
+                    # Get device info and serial number
+                    device_info = client.device_info
+                    if device_info is None:
+                        _LOGGER.error("No device info available from client")
+                        errors["base"] = "no_device_info"
+                    else:
+                        serial_number = device_info.get("SERIAL_NUMBER")
+                        if not serial_number:
+                            _LOGGER.error("No serial number found in device info")
+                            errors["base"] = "no_serial_number"
+                        else:
+                            await self.async_set_unique_id(serial_number)
+                            self._abort_if_unique_id_configured()
```

Okay so I was thinking about the config flow. The thing that is annoying about it, is that it's indented a lot. It makes it harder to understand and read and it makes me wonder, would it make sense to just raise an exception instead of returning the exception? WDYT?